src/redux-middleware/pushQueue.ts

```diff
@@ -108,27 +108,42 @@ const flushEdits = (pushQueue: PushBatch[]): Thunk<Promise<void>> => async (disp
 /** Pull all descendants of pending moves and dispatch existingThoughtMove to fully move. */
 const flushMoves = (pushQueue: PushBatch[]): Thunk => async (dispatch, getState) => {
 
-  // if there are pending thoughts that need to be moved, dispatch an action to be picked up by the pullQueue middleware which can load pending thoughts before dispatching another existingThoughtMove
-  const pendingMoves = pushQueue.map(batch => batch.pendingMoves || []).flat()
-  if (pendingMoves?.length) {
+  // if there are pending thoughts that need to be deleted, dispatch an action to be picked up by the pullQueue middleware which can load pending thoughts before dispatching another existingThoughtDelete
+  const descendantMoves = pushQueue.map(batch => batch.descendantMoves || []).flat()
+  const pendingPulls = pushQueue.map(batch => batch.pendingPulls || []).flat()
+
+  let maxDepth = Infinity
+  if (descendantMoves?.length) {
 
-    const pending: Index<Context> = keyValueBy(pendingMoves, ({ pathOld }) => {
+    const pending: Index<Context> = keyValueBy(descendantMoves, ({ pathOld }) => {
       const context = pathToContext(pathOld)
-      return {
-        [hashContext(context)]: context
-      }
+      // skip the pull for loaded descendants
+      return { [hashContext(context)]: context }
     })
 
     await dispatch(pull(pending, { maxDepth: Infinity }))
+    maxDepth = getDepth(getState(), pathToContext(descendantMoves[0].pathOld))
+  }
```

Comment:
Let's add a comment that clarifies that this pulls only source thoughts since that's important.